### PR TITLE
Add authorization

### PR DESCRIPTION
**Pull request by yosaddis**
# In this feature me and @gdumani  we added the following
 - Added cancancan gem to the project
 - Added the role property to the users model
 - Generated ability.rb and implement access to the 'admin' role
 - Added delete functionalities to the Post and comments based on user access

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -13,4 +13,8 @@ class User < ApplicationRecord
   def last_posts
     Post.where(author_id: id).order(created_at: :desc).limit(3)
   end
+
+  def admin?
+    role == 'admin'
+  end
```
Comment (MahmoudBakr23) [inline app/models/user.rb:19]: Kudos for the admin custom method 💯

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -25,6 +25,19 @@ def create
     end
   end
 
+  # def destroy
+  #   @post = Post.find(params[:post_id])
+  #   @post.destroy
+  # end
```
Comment (MahmoudBakr23) [inline app/controllers/posts_controller.rb]: - [ ] Please remove any unused code for a cleaner and more optimized code
- [ ] Also, as per project requirements, please consider using the CanCanCan gem built-in method `load_and_authorize_resource` inside your posts controller to invoke authorization for users while creating and/or deleting posts. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @yosaddis and @gdumani 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done installing and using the CanCanCan gem
- Good work creating the ability model and modifying the user model
- Descriptive PR and README
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers, and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @yosaddis  && @gdumani   👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Made changes as requested by the previous reviewer :+1: 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**